satellite/metainfo/metainfo.go

```diff
@@ -502,35 +528,69 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 		if allSizesValid {
 			redundancy, err := eestream.NewRedundancyStrategyFromProto(pointer.GetRemote().GetRedundancy())
 			if err != nil {
-				return Error.Wrap(err)
+				endpoint.log.Debug("pointer contains an invalid redundancy strategy", zap.Error(Error.Wrap(err)))
+				return rpcstatus.Error(rpcstatus.InvalidArgument, "invalid redundancy strategy; MinReq and/or Total are invalid")
 			}
 
 			expectedPieceSize := eestream.CalcPieceSize(pointer.SegmentSize, redundancy)
 			if expectedPieceSize != lastPieceSize {
-				return Error.New("expected piece size is different from provided (%v != %v)", expectedPieceSize, lastPieceSize)
+				errMsg := fmt.Sprintf("expected piece size is different from provided (%d != %d)", expectedPieceSize, lastPieceSize)
+				endpoint.log.Debug(errMsg)
```

Comment:
I'm not sure if this is necessarily better but I think you could also do `zap.Int64("Expected Piece Size", expectedPieceSize)` etc. instead of the fmt.Sprintf

Reply:
actually I guess errMsg is fine since I see it's being used by both the debug log and the returned error

Reply:
yes, the reason was for using the same message for the error and the `log.Debug`.

A structured logger is usually better than a normal one because it's easy to build reports and filtering messages, however this one is a debug level, so it should be less important.

Despite I would change it, my preference is to log with fields for the sake of the structured logger.

Reply:
Swapped to use log fields.

Reply:
So I saw that even though you swapped to use log fields here, there are other places in the PR that still use the fmt.Sprintf where the log fields could probably be used. I don't think that's a big deal enough to stop the PR but just wanted to point out in case it's better to be consistent.

Reply:
yes, I'd thought to recheck them, but after I forgot it.
The ones that I changed/added in this PR must be addressed and that's what I'm going to do.

Thanks for reporting it.

Reply:
I've done it.